swaybar/status_line.c

```diff
@@ -34,18 +35,35 @@ bool status_handle_readable(struct status_line *status) {
 	switch (status->protocol) {
 	case PROTOCOL_UNDEF:
 		errno = 0;
-		read_bytes = getline(&status->buffer,
-				&status->buffer_size, status->read);
-		if (errno == EAGAIN) {
-			clearerr(status->read);
-		} else if (errno) {
+		int available_bytes;
+		if (ioctl(status->read_fd, FIONREAD, &available_bytes) == -1) {
+			wlr_log(WLR_ERROR, "Unable to read status command output size");
 			status_error(status, "[error reading from status command]");
 			return true;
 		}
 
+		if ((size_t)available_bytes + 1 > status->buffer_size) {
+			// need room for leading '\0' too
+			status->buffer_size = available_bytes + 1;
+			status->buffer = realloc(status->buffer, status->buffer_size);
+		}
+		if (status->buffer == NULL) {
```

Comment:
This if block seems like it should be in the previous block

Reply:
I didn't want to nest the `if`s, and also thought that an extra check on `status->buffer` won't hurt anyone. Logging `errno` here could be misleading though. All in all, I'm not adamant on keeping it this way.